plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -327,11 +388,160 @@ describe('KubernetesProxy', () => {
 
     const response = await requestPromise;
 
+    expect(authTranslator.decorateClusterDetailsWithAuth).toHaveBeenCalledTimes(
+      0,
+    );
+    expect(response.status).toEqual(200);
+    expect(response.body).toStrictEqual({
+      kind: 'NamespaceList',
+      apiVersion: 'v1',
+      items: [],
+    });
+  });
+
+  it('returns a response with a localKubectlProxy auth provider configuration', async () => {
+    proxy = new KubernetesProxy({
+      logger: getVoidLogger(),
+      clusterSupplier: new LocalKubectlProxyClusterLocator(),
+      authTranslator: new NoopKubernetesAuthTranslator(),
+    });
+
+    worker.use(
+      rest.get('http://localhost:8001/api/v1/namespaces', (_req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.json({
+            kind: 'NamespaceList',
+            apiVersion: 'v1',
+            items: [],
+          }),
+        );
+      }),
+    );
+
+    permissionApi.authorize.mockReturnValue(
+      Promise.resolve([{ result: AuthorizeResult.ALLOW }]),
+    );
+
+    const router = Router();
+    router.use('/mountpath', proxy.createRequestHandler({ permissionApi }));
+    const app = express().use(router);
+
+    const requestPromise = request(app)
+      .get('/mountpath/api/v1/namespaces')
+      .set(HEADER_KUBERNETES_CLUSTER, 'local');
+
+    worker.use(rest.all(requestPromise.url, (req: any) => req.passthrough()));
+
+    const response = await requestPromise;
+
     expect(response.status).toEqual(200);
     expect(response.body).toStrictEqual({
       kind: 'NamespaceList',
       apiVersion: 'v1',
       items: [],
     });
   });
+
+  it('should return a 500 error if Backstage-Kubernetes-Cluster field isnt provided in request', async () => {
+    worker.use(
+      rest.get('https://localhost:9999/api/v1/namespaces', (_req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.json({
+            kind: 'NamespaceList',
+            apiVersion: 'v1',
+            items: [],
+          }),
+        );
+      }),
+    );
+
+    permissionApi.authorize.mockReturnValue(
+      Promise.resolve([{ result: AuthorizeResult.ALLOW }]),
+    );
+
+    clusterSupplier.getClusters.mockResolvedValue([
+      {
+        name: 'cluster1',
+        url: 'https://localhost:9999',
+        authProvider: 'googleServiceAccount',
+      },
+    ] as ClusterDetails[]);
+
+    authTranslator.decorateClusterDetailsWithAuth.mockResolvedValue({
+      name: 'cluster1',
+      url: 'https://localhost:9999',
+      serviceAccountToken: 'tokenA',
+      authProvider: 'googleServiceAccount',
+    } as ClusterDetails);
+
+    const router = Router();
+    router.use('/mountpath', proxy.createRequestHandler({ permissionApi }));
+    const app = express().use(router);
```

Comment:
To get the right error mapping, I think we could add something like this
```suggestion
    const router = Router();
    router.use('/mountpath', proxy.createRequestHandler({ permissionApi }));
    router.use(errorHandler());
    const app = express().use(router);
```
where `errorHandler` is the function from `@backstage/backend-common` documented [here](https://backstage.io/docs/reference/backend-common.errorhandler/)

Reply:
Done